src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -312,6 +291,38 @@ public void persistentQueueSync(Iterable<Long> diskQueueOfIds, CloudMessageSpool
         nextId.set(highestId + 1);
     }
 
+    /**
+     * Extract message ids from the persistenceSpool plugin's disk database and insert the message
+     * ids into queueOfMessageId, this function is only used in Disk storage mode. If Sync fails midway,
+     * we continue anyway with that DiskSpooler. If we fail to get all Message IDs from Disk Spooler Database,
+     * we default to InMemory spooler.
+     *
+     * @param diskSpooler   the disk spooler instance for syncing messages
+     */
+    public void executeQueueSync(CloudMessageSpool diskSpooler) {
+        try {
+            persistentQueueSync(diskSpooler.getAllMessageIds(), diskSpooler);
```

Comment:
why are we doing this? Isn't this effectively just copying from diskSpooler into diskSpooler? What is this accomplishing

Reply:
We maintain a Queue which has all message IDs that we mean to send to Cloud. When publishing or adding messages to spool, we use this queue to figure out what to publish next or if we can add more messages to spool.
We are performing this sync to basically refill that queue, with the messages that we see in persistent storage

Reply:
Thought of a case just now, where we only restart the DiskSpooler component(which calls ExecuteSync) but not Nucleus, the queue would already contain these message IDs in that case. Adding a check for that so that we don't add duplicate message IDs to main queue while syncing.

Reply:
instead of adding checks, is it possible to dedupe? use an ordered set for example/map